arch/arm/core/aarch32/cortex_m/reset.S

```diff
@@ -63,6 +66,13 @@ SECTION_SUBSEC_FUNC(TEXT,_reset_section,z_arm_reset)
  */
 SECTION_SUBSEC_FUNC(TEXT,_reset_section,__start)
 
+#if defined(CONFIG_DEBUG_THREAD_INFO)
+    /* Clear z_sys_post_kernel flag for RTOS aware debuggers */
+    movs.n r0, #0
+    ldr r1, =z_sys_post_kernel
+    str r0, [r1]
```

Comment:
`z_sys_post_kernel`'s type is `bool`, which is 1-byte for most toolchains..

```suggestion
    strb r0, [r1]
```

Reply:
Do you know of a way to detect the actual size in a `#if`?  It would be nice to choose the correct instruction automatically, rather than one that works for "most" toolchains.
Originally I had it as a word store because this happens before `.bss` is cleared and `.data` is copied, so writing extra bytes to 0 isn't overriding other variables.

Reply:
> Do you know of a way to detect the actual size in a #if? 

No, there is no universal way of doing that only using the C preprocessor.

> It would be nice to choose the correct instruction automatically

The only way to achieve that would be to implement a function that does this in C and call it here.

> writing extra bytes to 0 isn't overriding other variables.

Even so, it is not really a sane thing to do. Also, there is no guarantee that `z_sys_post_kernel` will not be the last variable in a section and that the section will be word-aligned.